### PR TITLE
[Add] configs for QDTrack on TAO dataset.

### PR DESCRIPTION
**Pull request by Seerkfang**

### CODE REVIEW

**Review by GT9505 — COMMENTED** [review]
Do we need to tracker for TAO dataset?

File: configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# model settings
+_base_ = [
+    './qdtrack_faster-rcnn_r101_fpn_24e_lvis.py',
```
Comment (GT9505) [inline configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py:3]: inherit from `'./qdtrack_faster-rcnn_r101_fpn_24e_lvis.py'` is enough

File: configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    dict(type='SeqRandomFlip', share_params=True, flip_ratio=0.5),
+    dict(type='SeqNormalize', **img_norm_cfg),
+    dict(type='SeqPad', size_divisor=32),
+    dict(type='MatchInstances', skip_nomatch=True),
+    dict(
+        type='VideoCollect',
+        keys=['img', 'gt_bboxes', 'gt_labels', 'gt_match_indices'],
+    ),
+    dict(type='SeqDefaultFormatBundle', ref_prefix='ref'),
+]
```
Comment (GT9505) [inline configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py:28]: Can these be reused?

Comment (Seerkfang) [reply configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py:28]: The train data source and dataset type is different, so we have to rewrite here.

File: mmtrack/models/trackers/tao_tracker.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+import torch
+from mmdet.core import bbox_overlaps
+
+from mmtrack.core import embed_similarity, imshow_tracks
+from ..builder import TRACKERS
+from .base_tracker import BaseTracker
+
+
+@TRACKERS.register_module()
+class TaoTracker(BaseTracker):
```
Comment (GT9505) [inline mmtrack/models/trackers/tao_tracker.py:14]: Rename it to `QuasiDenseTAOTracker`.
And rename `QuasiDenseEmbedTracker` to `QuasiDenseMOTTracker`

Comment (GT9505) [inline mmtrack/models/trackers/tao_tracker.py:14]: inherit from qdtrack tracker of MOT17

File: mmtrack/models/trackers/tao_tracker.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+        # init tracklets
+        new_inds = (ids == -1) & (bboxes[:, 4] > self.init_score_thr).cpu()
+        num_news = new_inds.sum()
+        ids[new_inds] = torch.arange(
+            self.num_tracks, self.num_tracks + num_news, dtype=torch.long)
+        self.num_tracks += num_news
+
+        self.update(ids, bboxes, labels, embeds, frame_id)
+
+        # show results of gt tracks and predicted tracks
```
Comment (GT9505) [inline mmtrack/models/trackers/tao_tracker.py:249]: delete the visualization in here

File: mmtrack/models/trackers/tao_tracker.py [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+            img = imshow_tracks(
+                img,
+                bboxes[idsw_inds].numpy(),
+                labels[idsw_inds].numpy(),
+                ids[idsw_inds].numpy(),
+                color='cyan',
+                show=False,
+                out_file=metas.out_file)
+
+        return bboxes, labels, ids
```
Comment (GT9505) [inline mmtrack/models/trackers/tao_tracker.py:368]: where is the unittest?

File: configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py (hunk 2 above)
Comment (GT9505) [reply configs/mot/qdtrack/qdtrack_faster-rcnn_r101_fpn_12e_tao.py:25]: But the `train_pipeline` can be used?

**Review by GT9505 — COMMENTED** [review]
The PR is not ready for review.

File: mmtrack/models/trackers/tao_tracker.py (hunk 3 above)
Comment (Seerkfang) [reply mmtrack/models/trackers/tao_tracker.py:14]: ok, but the QuasiDenseEmbedTracker is used by datasets other than TAO, like bdd and mot17, should we change the name to mot?

Comment (Seerkfang) [reply mmtrack/models/trackers/tao_tracker.py:14]: These two trackers have the same function but total different implementations (memo / update / track). I think there is no need to inherit from the former tracker.

Comment (GT9505) [reply mmtrack/models/trackers/tao_tracker.py:14]: Then, rename `TaoTracker` to `QuasiDenseTAOTracker`, and `QuasiDenseEmbedTracker` to ``QuasiDenseTracker

Comment (GT9505) [reply mmtrack/models/trackers/tao_tracker.py:14]: Ok, keep it unchanged.

Comment (GT9505) [reply mmtrack/models/trackers/tao_tracker.py:14]: And remember to add more comments in the docstring of `QuasiDenseTAOTracker` and `QuasiDenseTracker` to explain that the `QuasiDenseTAOTracker` is for TAO and `QuasiDenseTracker` is for MOT Challenge and BDD100K.

**Review by GT9505 — COMMENTED** [review]
This PR is blocked.

**Pull request closed without merging**